test/test_data_serialization.py

```diff
@@ -214,6 +214,19 @@ def test_encoder_datetime(self):
             decoded = json.loads(encoded, cls=RuntimeDecoder)
             self.assertEqual(decoded, obj)
 
+    def test_encoder_instruction(self):
+        """Test encoding and decoding instructions"""
+        subtests = (
+            {"instruction": CXGate()},
+            {"instruction": PhaseGate(theta=1)},
+            {"instruction": U2Gate(phi=1, lam=1)},
```

Comment:
It'd probably be good to test parameterized instructions too, just to make sure those are handled correctly. If you're really handling submission of standalone instruction objects to runtimes (I still don't know what the 
use case for this could be) having unbound parameters on the instruction seems worth testing here. Something like:

```suggestion
            {"instruction": CXGate()},
            {"instruction": PhaseGate(theta=1)},
            {"instruction": U2Gate(phi=1, lam=1)},
            {"instruction": U2Gate(phi=Parameter('phi'), lam=Parameter('lambda'))},
```

and maybe one with a `ParameterVector` element and a `ParameterExpression` too.